Automattic/pocket-casts-android
# Provide MiniPlayerState to save/restore isPlaying state for MiniPlayer View

## Description
Provide `MiniPlayerState` to restore `playing` value after `MiniPlayer` view is recreated. Currently `MiniPlayer` view after recreation always starts with the state `playing=state` even if podcast is playing. This may lead to different glitches with animating play/pause state (like that one described in https://github.com/Automattic/pocket-casts-android/issues/523). I was considering using [BaseSavedState](https://developer.android.com/reference/android/view/View.BaseSavedState) instead of plain `Parcelable` but I think in this case it's fine since `MiniPlayer` won't be extended by other classes.

Fixes #523 

## Testing Instructions
1. Start playing a podcast
2. Switch themes in the app
3. Play button should not switch back and forth a few times between play and pause 

## Screenshots or Screencast 
https://user-images.githubusercontent.com/529635/199326831-6208dbd1-e1e7-4e26-8c81-4207869f53d8.mp4

## Checklist
 
#### I have tested any UI changes...
